src/components/button-group/button-group.component.ts

```diff
@@ -28,6 +28,40 @@ export default class SlButtonGroup extends ShoelaceElement {
    */
   @property() label = '';
 
+  private throttleTimer: ReturnType<typeof setTimeout> | null = null;
+  private throttledSlotChanged = () => {
+    // Don't ask me why, but tests fail without an initial setTimeout. I also tried using async / await this.updateComplete.
+    // Throttle seems more appropriate than debouncing.
+    setTimeout(() => {
+      if (this.throttleTimer === null) {
+        this.handleSlotChange();
+        this.throttleTimer = setTimeout(() => {
+          this.throttleTimer = null;
+        }, 30);
+      }
+    });
+  };
```

Comment:
This doesn't feel very good. Can we switch to data attributes instead of classes and avoid this logic altogether?

Reply:
Probably, let me see what React decides to do.